pkg/networkservice/up/server.go

```diff
@@ -99,8 +100,15 @@ func (u *upServer) Close(ctx context.Context, conn *networkservice.Connection) (
 }
 
 func (u *upServer) init(ctx context.Context) error {
-	u.Do(func() {
-		u.initErr = initFunc(ctx, u.vppConn)
-	})
-	return u.initErr
+	if !u.inited.Load() {
```

Comment:
This looks not thread-safe. A few goroutines could read `u.inited` as a false and block on the u.initMutex.

Reply:
Probably we could add something like this https://go.dev/play/p/f2J7aioP59W

Reply:
@edwarnicke Thoughts?

Reply:
As an alternative, we also could use a personal Executor.

Reply:
@denis-tingaikin 
I like the idea about extended sync.Once and add it to sdk/tools.
Prepared PR: https://github.com/networkservicemesh/sdk/pull/1224

Reply:
made it inline